content/200-concepts/100-components/02-prisma-client/062-computed-fields.mdx

```diff
@@ -10,6 +10,81 @@ Computed fields allow you to derive a new field based on existing data. A common
 
 </TopBlock>
 
+## Computed fields with Prisma Client extensions
+
+> Prisma Client extensions are currently in [Preview](/about/prisma/releases#preview).
+
+The following example illustrates how to create a [Prisma Client extension](/concepts/components/prisma-client/client-extensions) that adds a virtual/ computed field to a model in a Prisma schema. The example adds a `fullName` property that is computed at runtime.
+
+<TabbedContent tabs={[<FileWithIcon text="Prisma Client extension" icon="code"/>, <FileWithIcon text="Example query" icon="code"/>,<FileWithIcon text="Prisma schema" icon="file"/>]}>
+
+<tab>
+
+```tsx
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient().$extends({
+  result: {
+    user: {
+      fullName: {
+        needs: { firstName: true, lastName: true },
+        compute(user) {
+          return `${user.firstName} ${user.lastName}`
+        },
+      },
+    },
+  },
+})
+```
+
+</tab>
+
+<tab>
+
+```ts
+const user = await prisma.user.findMany({
```

Comment:
it's nice that you're showing an example query here, but I don't find it too helpful without further context and hidden in the tabs. how is it going to help someone reading this page understand computed fields better?

it would be more helpful IMO if:
- the query wasn't hidden in a tab
- we showed the actual resulting structure that includes the new field and a short explanation